src/custom-element.js

```diff
@@ -117,96 +117,116 @@ export function stubElements(win) {
 
 
 /**
+ * Determines the layout of the element from the layout attribute, and if
+ * not specified, by considering hasNaturalDimensions() and the width,
+ * height, and sizes attributes.
+ * @param {!AmpElement} element
+ * @param {string} widthAttr
+ * @param {string} heightAttr
+ * @return {!Layout}
+ */
+function determineLayout(element, widthAttr, heightAttr) {
+  const layoutAttr = element.getAttribute('layout');
+  if (layoutAttr) {
+    const parsed = parseLayout(layoutAttr);
+    if (!parsed) {
+      throw new Error('Layout attr value invalid: ' + layoutAttr);
+    }
+    return parsed;
+  }
+  if (hasNaturalDimensions(element.tagName)) {
```

Comment:
This is not correct. The element could have `width=auto` and that would make it FIXED_HEIGHT.